modules/src/ics03_connection/connection.rs

```diff
@@ -25,6 +29,34 @@ impl ConnectionEnd {
             versions: validate_versions(versions).map_err(|e| Kind::InvalidVersion.context(e))?,
         })
     }
+
+    pub fn set_state(&mut self, new_state: State) {
+        self.state = new_state;
+    }
+
+    pub fn from_proto_connection(pc: proto_connection::ConnectionEnd) -> Result<Self, Error> {
+        // The Counterparty field is an Option, may be missing.
+        match pc.counterparty {
+            Some(cp) => {
+                let mut conn = ConnectionEnd::new(
+                    ClientId::from_str(&pc.client_id).unwrap(),
```

Comment:
Could we get rid of the `unwrap()`s? If the user issues a query command and gets back a few pages of traces it does not help. I had this comment earlier but I guess it got lost. Try to query with non-exsiting connection id.

Reply:
I'd like to make this into a From trait, if possible.

Reply:
Also I guess this should be verifying all the fields, not just 2? And should we really have an error per field or maybe start with a more general InvalidConnectionEnd error of some kind and include what's missing as a string? Not sure ...

Reply:
> I'd like to make this into a From trait, if possible.

Either option could work, the question is if we want to keep validation inside the `from` or outside. The current code uses the `FromStr` trait, with the signature `fn from_str(s: &str) -> Result<Self, Self::Err>` so we can do validation inside there and return error if validation fails. With `From`, the sig is `fn from(T) -> Self` so we'd have to do validation after calling `from`. Is it generally speaking, or specifically in this case, better to do validation outside of `from`?

> Also I guess this should be verifying all the fields, not just 2?

Agree this deserves a discussion to find a more principled approach.